src/main/java/org/jabref/gui/search/GlobalSearchBar.java

```diff
@@ -202,6 +203,15 @@ public void actionPerformed(ActionEvent e) {
                 searchField,
                 currentResults
         );
+
+        // Clears search on ESC & select first entry, if available
+        searchField.setOnKeyPressed((event -> {
+            if (event.getCode() == KeyCode.ESCAPE) {
```

Comment:
The "Clear search" shortcut is customizable. Please use https://github.com/JabRef/jabref/blob/667525afe3d36ea1636ce814aa81c2d702e009b9/src/main/java/org/jabref/gui/keyboard/KeyBinding.java#L14
to check for the correct key (using a KeyBindingRepository).

Reply:
Just take a look at the setupKeyBindings in the EntryEditor, we use that code in several other classes, too.